### PR TITLE
Perf/increase performance when spy is disabled

### PR DESCRIPTION
**Pull request by Ascenio**
## What
This PR increases [mobx](https://pub.dev/packages/mobx) performance by avoiding creating Strings and DateTime objects when _spying_ is off.

## Benchmarking code
```dart
import 'package:mobx/mobx.dart';

part 'main.g.dart';

class CounterStore = _CounterStoreBase with _$CounterStore;

abstract class _CounterStoreBase with Store {
  @observable
  int counter = 0;

  @action
  void increment() {
    counter++;
  }
}

void main(List<String> args) {
  final counter = CounterStore();
  final watch = Stopwatch()..start();
  for (var i = 0; i < 1000000; i++) {
    counter.increment();
  }
  watch.stop();
  print(watch.elapsedMilliseconds);
}
```

## Results

In my laptop, it went down from **~530ms** to shocking **~250ms** :laughing:.

You can also take a look at the bottom-up info taken from [Dart DevTools](https://dart.dev/tools/dart-devtools) bellow. Notice how our bottleneck reduced.

<table>
<tr>
<th>Before</th>
<th>After</th>
</tr>
<tr>
<td>
</td>
<td>
</td>
</tr>
</table>

### CODE REVIEW

**Comment by Ascenio** [comment]
Hey, @pavanpodila. Could you, please, re-run the CI? #722 has it fixed it! ;D

File: mobx/lib/src/core/atom_extensions.dart [hunk 1]
```diff
@@ -10,11 +10,13 @@ extension AtomSpyReporter on Atom {
     context.spyReport(ObservableValueSpyEvent(this,
         newValue: newValue, oldValue: oldValue, name: name));
 
+    final actionName = context.isSpyEnabled ? '${name}_set' : null;
```
Comment (pavanpodila) [inline mobx/lib/src/core/atom_extensions.dart:13]: `actionName` should be set to `name` when spy is disabled

**Review by pavanpodila — CHANGES_REQUESTED** [review]
just a minor change, rest looks good

File: mobx/lib/src/core/atom_extensions.dart (hunk 1 above)
Comment (pavanpodila) [reply mobx/lib/src/core/atom_extensions.dart:13]: Also @Ascenio, there are some analyser errors: https://github.com/mobxjs/mobx.dart/runs/4590667181?check_suite_focus=true

Comment (Ascenio) [reply mobx/lib/src/core/atom_extensions.dart:13]: About the analyzer errors, they were are already in mobx's master. PR #722 fixed it.

Comment (Ascenio) [reply mobx/lib/src/core/atom_extensions.dart:13]: Do you mean like this?
```suggestion
    final actionName = context.isSpyEnabled ? '${name}_set' : name;
```

Comment (pavanpodila) [reply mobx/lib/src/core/atom_extensions.dart:13]: Yes

File: mobx/lib/src/api/observable_collections/observable_map.dart [hunk 2]
```diff
@@ -73,7 +73,8 @@ class ObservableMap<K, V>
     _context.enforceReadPolicy(_atom);
 
     _atom.reportObserved();
-    return _map[key as K?];
+    final castedKey = key as K?;
```
Comment (pavanpodila) [inline mobx/lib/src/api/observable_collections/observable_map.dart:76]: Please update from master to remove these lines. We are going with a simpler approach

**Review by pavanpodila — CHANGES_REQUESTED** [review]

File: mobx/lib/src/api/observable_collections/observable_map.dart (hunk 2 above)
Comment (pavanpodila) [reply mobx/lib/src/api/observable_collections/observable_map.dart]: I've committed a fix for this

**Pull request merged**